feat(images): add support for list procurement images

bfrgoncalves opened the pull request:
Resolves: https://lifebit.atlassian.net/browse/LP-60948

Adds support for listing images associated with organisations of a given procurement. 

Proof of AC
Can list without and with pagination
<img width="1005" height="1206" alt="Screenshot 2025-07-25 at 11 22 57" src="https://github.com/user-attachments/assets/ede15d2b-3c34-4051-ac8a-a274baf1fd8f" />
<img width="890" height="529" alt="Screenshot 2025-07-25 at 11 23 11" src="https://github.com/user-attachments/assets/50e1d5de-6697-49e4-90ff-5d1e193f6c7f" />

dapineyro reviewed (changes requested):
Thanks @bfrgoncalves for this great contribution! Code-wise, I have only some minor requests. Apart from them:
- Could you update `cloudos-cli/_version.py` to a new version: `2.46.0` ?
- Could you add an entry in CHANGELOG.md for this new version?
- I've tested without adding `--procurement_id` and it does not ask for it as required (I left a comment where you should modify for make it work if you have a profile without this value) and also we don't have a specific error for when the procurement_id is not correct. Would it be possible (and easy, so it can be added in the scope of this PR) to capture that specific error and message the user so they know what is failing?
<img width="704" height="155" alt="Screenshot 2025-07-25 at 20 15 41" src="https://github.com/user-attachments/assets/a34ce55e-4cd4-47e1-b2b3-1647c2eab75b" />

bfrgoncalves commented on cloudos_cli/__main__.py:
```diff
@@ -3497,6 +3520,88 @@ def link(ctx, path, apikey, cloudos_url, project_name, workspace_id, session_id,
+        'apikey': True,
+        'workspace_id': False,
+        'workflow_name': False,
+        'project_name': False
```
Added now 👍 thx for the suggestion.
It now gives the following error message:
<img width="930" height="381" alt="Screenshot 2025-07-28 at 14 11 19" src="https://github.com/user-attachments/assets/5c3e5dec-7e31-4ff5-b3bb-02806758aaea" />

bfrgoncalves commented:
@dapineyro I've addressed your comments. 
Regarding adding a bad request reason, that would be modifying the correct implementation that exists on other integrations. Currently it is just exposing the error according to the response. If that is to change, I think it should be on other more general scope.

dapineyro commented on README.md:
````diff
@@ -1144,6 +1144,91 @@ Please, note that in the above example a preconfigured profile has been used. If
+    --limit 25
+```
+
+#### Using profiles
+
+You can also use profiles to simplify the command. First, configure a profile with your procurement details:
+
+```bash
+cloudos configure set-profile procurement-profile \
+    --apikey $MY_API_KEY \
+    --cloudos-url $CLOUDOS \
+    --procurement-id $PROCUREMENT_ID
+```
+
+Then use the profile:
+
+```bash
+cloudos procurement images ls --profile procurement-profile
+```
````
```suggestion
```
I'd suggest to get rid of all this section. It is essentially incorrect (`cloudos configure set-profile` is not a command, it should be `cloudos configure --profile procurement-profile` and takes no argument). If you want, you could add a small entry informing the user `--procurement-id` is now a configurable parameter

bfrgoncalves replied:
Added just a small section saying that procurement-id can be configured via profile.

dapineyro reviewed (dismissed):
LGTM!
